drivers/abp2/abp2.c

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (C) 2024 CNRS, France
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_abp2
+ * @{
+ *
+ * @file
+ * @brief       Honeywell ABP2 series pressure and temperature sensor driver
+ *
+ * @author      David Picard <david.picard@clermont.in2p3.fr>
+ * @}
+ */
+
+
+#include <assert.h>
+#include "abp2.h"
+#include "abp2_params.h"
+#include "periph/spi.h"
+#include "periph/gpio.h"
+#include "ztimer.h"
+#include <errno.h>
+#include "debug.h"
+
+#define ENABLE_DEBUG 0
+#define MAX_LOOPS_TIMEOUT   (10)            /**< Timeout (ms). */
+#define DEV (dev->params.spi)
+#define ABP2_CMD_MEAS       (0xAA)          /**< Start a measurement. */
+#define ABP2_CMD_NOP        (0xF0)          /**< NOP command. */
+#define ABP2_RX_BUF_LEN     (8)             /**< Length of the receive buffer. */
+
+
```

Comment:
```suggestion

```

sorry but our static tests don't like double empty line  even though they might help the code structure a little bit